libnm-core/nm-connection.c

```diff
@@ -1253,15 +1253,37 @@ _normalize_ip_tunnel_wired_setting (NMConnection *self, GHashTable *parameters)
 }
 
 static gboolean
-_normalize_sriov_vf_order (NMConnection *self, GHashTable *parameters)
+_normalize_sriov (NMConnection *self, GHashTable *parameters)
 {
 	NMSettingSriov *s_sriov;
+	gboolean ret = FALSE;
 
 	s_sriov = NM_SETTING_SRIOV (nm_connection_get_setting (self, NM_TYPE_SETTING_SRIOV));
 	if (!s_sriov)
-		return FALSE;
+		return ret;
+
+	/* 1. set the enable property if there are VFs configured */
+	if (!nm_setting_sriov_get_enable (s_sriov)) {
+		if (   nm_setting_sriov_get_total_vfs (s_sriov)
+		    || nm_setting_sriov_get_num_vfs (s_sriov)) {
+			g_object_set (G_OBJECT (s_sriov),
+			              NM_SETTING_SRIOV_ENABLE, TRUE,
+			              NULL);
+			ret = TRUE;
+		}
+	}
+
 
-	return _nm_setting_sriov_sort_vfs (s_sriov);
+	/* 2. remove the setting if everything is disabled */
+	if (!nm_setting_sriov_get_enable (s_sriov)) {
```

Comment:
setting's `verify()`  must mirror the normalization steps, which is not the case for this normalization.

Also, it seems questionable, to have a property whose only value is always enforced via normalization. If you don't want to keep a sriov setting that is disabled, then there should not be a new sriov.enable property but the user should remove the SRIOV section entirely. Maybe that's anyway a better solution.

Reply:
The problem with removing the sriov setting entirely is that it is not possible to do it from nmcli (is it?), where users can only control single properties.

Reply:
there is with `nmcli connection edit`. If it isn't with command line, then that is a missing feature (of not very high priority).